mslice/scripting/__init__.py

```diff
@@ -22,15 +21,57 @@ def generate_script(ws_name, filename=None, plot_handler=None, window=None):
 
 def preprocess_lines(ws_name, plot_handler):
     from mslice.plotting.plot_window.cut_plot import CutPlot
+    script_lines = []
 
     if isinstance(plot_handler, CutPlot) and len(get_cut_plotter_presenter()._cut_cache) > 1:
-        script_lines = []
-        for workspace_name in get_cut_plotter_presenter()._cut_cache:
-            ws = get_workspace_handle(workspace_name).raw_ws
-            script_lines += ["\nws_{} = mc.".format(workspace_name.replace(".", "_"))] + cleanup(
-                GeneratePythonScript(ws).split('\n'))
+        cut_cache = get_cut_plotter_presenter()._cut_cache
+        cache_list = get_cut_plotter_presenter()._cut_cache_list
+        for workspace_name in cut_cache:
+            if any([workspace_name.replace(".", "_") == cut.workspace_name for cut in cache_list]):
+                ws = get_workspace_handle(workspace_name).raw_ws
+                script_lines += generate_script_lines(ws, workspace_name)
     else:
         ws = get_workspace_handle(ws_name).raw_ws
-        script_lines = ["\nws = mc."] + cleanup(GeneratePythonScript(ws).split('\n'))
+        script_lines += generate_script_lines(ws, ws_name)
 
     return script_lines
+
+
+def generate_script_lines(raw_ws, workspace_name):
+    lines = []
+    ws_name = workspace_name.replace(".", "_")
+    alg_history = raw_ws.getHistory().getAlgorithmHistories()[::-1]
```

Comment:
There is a function in python called `reversed` which is handy for doing this:
```python
alg_history = raw_ws.getHistory().getAlgorithmHistories()
for algorithm in reversed(alg_history)
...
```
It makes it easier to read as it's easy to forget what `[::-1]` does.